app/Http/Controllers/Common/CronController.php

```diff
@@ -438,6 +438,70 @@ public function postRenewalNotify()
         }
     }
 
+    /**
+     * Deletes old invoices based on specified criteria.
+     *
+     * This function checks if invoices should be deleted, retrieves old invoices,
+     * and deletes invoices that meet specific conditions.
+     *
+     * @return void
+     */
+    public function invoicesDeletion()
+    {
+        if ($this->shouldDeleteInvoices()) {
```

Comment:
Don't nest your loop inside the codition, if(false) just return and then if it  is true it will execute your code